Source/hydro/Castro_mol_hydro.cpp

```diff
@@ -289,23 +288,29 @@ Castro::construct_mol_hydro_source(Real time, Real dt, MultiFab& A_update)
 
 #if AMREX_SPACEDIM >= 2
             // construct the face-center interface states q_fc
-            AMREX_PARALLEL_FOR_4D(nbx, NQ, i, j, k, n, {
-                bool test = (n == QGC) || (n == QTEMP);
+            const int* lo_bc = phys_bc.lo();
```

Comment:
`phys_bc` is sized `AMREX_SPACEDIM` so we need to avoid dereferencing it outside those bounds. There are a few other instances of that below.